File: backend/routes/resignation.go
package routes

import (
	"oa-system/handlers"

	"github.com/gin-gonic/gin"
)

func resignationRoutes(rg *gin.RouterGroup) {
	rg.GET("/resignations", handlers.GetResignations)
	rg.GET("/resignations/:id", handlers.GetResignation)
	rg.POST("/resignations", handlers.CreateResignation)
	rg.PUT("/resignations/:id", handlers.UpdateResignation)
	rg.PUT("/resignations/:id/submit", handlers.SubmitResignation)
	rg.PUT("/resignations/:id/withdraw", handlers.WithdrawResignation)
	rg.PUT("/resignations/:id/approve", handlers.ApproveResignation)
	rg.PUT("/resignations/:id/cancel-approve", handlers.CancelApproveResignation)
	rg.DELETE("/resignations/:id", handlers.DeleteResignation)
}

File: frontend/src/api/resignation.js
import http from './http'

export const getResignations = (params) => http.get('/resignations', { params })
export const getResignation = (id) => http.get(`/resignations/${id}`)
export const createResignation = (data) => http.post('/resignations', data)
export const updateResignation = (id, data) => http.put(`/resignations/${id}`, data)
export const deleteResignation = (id) => http.delete(`/resignations/${id}`)
export const submitResignation = (id, data) => http.put(`/resignations/${id}/submit`, data)
export const withdrawResignation = (id) => http.put(`/resignations/${id}/withdraw`, {})
export const approveResignation = (id, data) => http.put(`/resignations/${id}/approve`, data)
export const cancelApproveResignation = (id) => http.put(`/resignations/${id}/cancel-approve`, {})
